src/email/email.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@nestjs/common'
+import { EmailData, MailgunService } from './mailgun.service'
+import {
+  getSetPasswordEmailContent,
+  getBasicEmailContent,
+  getRecoverPasswordEmailContent,
+} from './util/content.util'
+import { User } from '@prisma/client'
+
+const APP_BASE = process.env.CORS_ORIGIN as string
+
+type SendEmailInput = {
+  to: string
+  subject: string
+  message: string
+  from?: string
+}
+
+type SendTemplateEmailInput = {
+  to: string
+  subject: string
+  template: string
+  variables?: object
+  from?: string
+  cc?: string
+}
+
+@Injectable()
+export class EmailService {
+  constructor(private mailgun: MailgunService) {}
+
+  async sendEmail({ to, subject, message, from }: SendEmailInput) {
+    return await this.sendEmailWithRetry({
+      from: from || 'GoodParty.org <noreply@goodparty.org>',
+      to,
+      subject,
+      text: message,
+      html: getBasicEmailContent(message, subject),
+    })
+  }
+
+  async sendTemplateEmail({
+    to,
+    subject,
+    template,
+    variables = {},
+    from,
+    cc,
+  }: SendTemplateEmailInput) {
+    const data: EmailData = {
+      from: from || 'GoodParty.org <noreply@goodparty.org>',
+      to,
+      subject,
+      template,
+      variables: {
+        appBase: APP_BASE,
+        ...variables,
+      },
+    }
+
+    if (cc) {
+      data.cc = cc
+    }
+
+    return await this.sendEmailWithRetry(data)
+  }
+
+  async sendRecoverPasswordEmail(user: User) {
+    const { firstName, lastName, email, passwordResetToken } = user
+    const encodedEmail = email.replace('+', '%2b')
+    const link = encodeURI(
+      `${APP_BASE}/reset-password?email=${encodedEmail}&token=${passwordResetToken}`,
```

Comment:
If the user can be looked up by the token, then why bother encoding putting the `email` in the link?

I'm not seeing it being used anywhere here in the `/reset-password` endpoint in the `UserService` 🤔 

https://github.com/thegoodparty/gp-api/pull/23/files?diff=split&w=1#diff-090521db6f185ea4dde00b85584328375154804ef0e7b0b5a3bd4030fa91d322R49-R59

Reply:
Ah.. ok. 

Guess we can clean it up later on then.